### PR TITLE
Validate MSIDTokenResponse instead of json dictionary in MSIDTokenResponseValidator

### PR DESCRIPTION
**Pull request by antrix1989**

### CODE REVIEW

File: IdentityCore/src/requests/MSIDInteractiveTokenRequest.m [hunk 1]
```diff
@@ -229,18 +230,18 @@ - (void)acquireTokenWithCode:(NSString *)authCode
                                                                                                            codeVerifier:self.webViewConfiguration.pkce.codeVerifier
                                                                                                                authCode:authCode
                                                                                                           homeAccountId:self.authCodeClientInfo.accountIdentifier];
+    tokenRequest.responseSerializer = [[MSIDAADTokenResponseSerializer alloc] initWithOauth2Factory:self.oauthFactory];
```
Comment (oldalton) [inline IdentityCore/src/requests/MSIDInteractiveTokenRequest.m]: what if it's not AAD? Should we maybe set token response serializer in factory itself where MSIDAuthorizationCodeGrantRequest is created?

Comment (antrix1989) [reply IdentityCore/src/requests/MSIDInteractiveTokenRequest.m]: Agree, changed.

File: IdentityCore/src/network/response_serializer/MSIDHttpResponseSerializer.m [hunk 2]
```diff
@@ -30,7 +30,20 @@ - (id)responseObjectForResponse:(NSHTTPURLResponse *)httpResponse
                         context:(id <MSIDRequestContext>)context
                           error:(NSError **)error
 {
-    return data;
+    id result;
```
Comment (unpluggedk) [inline IdentityCore/src/network/response_serializer/MSIDHttpResponseSerializer.m]: nit: do we even need this variable?

File: IdentityCore/src/network/response_serializer/MSIDResponseSerialization.h [hunk 3]
```diff
@@ -25,15 +25,18 @@
 
 @protocol MSIDRequestContext;
 
+NS_ASSUME_NONNULL_BEGIN
```
Comment (unpluggedk) [inline IdentityCore/src/network/response_serializer/MSIDResponseSerialization.h]: nit: do we need this? seems like eveything is nullable, so perhaps simply remove this with nullability identifiers?

File: IdentityCore/tests/integration/MSIDHttpRequestIntegrationTests.m [hunk 4]
```diff
@@ -108,9 +109,14 @@ - (void)testUrlRequest_byDefaultIsNil
     XCTAssertNil(self.request.urlRequest);
 }
 
-- (void)testRequest_byDefaultUseMSIDJsonResponseSerializer
+- (void)testRequest_byDefaultUseMSIDHttpResponseSerializerWithMSIDJsonResponsePreprocessor
 {
-    XCTAssertTrue([self.request.responseSerializer isKindOfClass:MSIDJsonResponseSerializer.class]);
+    XCTAssertTrue([self.request.responseSerializer isKindOfClass:MSIDHttpResponseSerializer.class]);
+    if ([self.request.responseSerializer isKindOfClass:MSIDHttpResponseSerializer.class])
```
Comment (unpluggedk) [inline IdentityCore/tests/integration/MSIDHttpRequestIntegrationTests.m]: Curious on why we have a if here,
shouldn't this always the case as we are checking this very case?

File: IdentityCore/src/network/response_serializer/MSIDHttpResponseSerializer.m (hunk 2 above)
Comment (antrix1989) [reply IdentityCore/src/network/response_serializer/MSIDHttpResponseSerializer.m]: I prefer to keep it. In my opinion, it looks weird if we have returns inside "if" but there is no return at the end of the method.

File: IdentityCore/src/network/response_serializer/MSIDResponseSerialization.h (hunk 3 above)
Comment (antrix1989) [reply IdentityCore/src/network/response_serializer/MSIDResponseSerialization.h]: removed it.

File: IdentityCore/tests/integration/MSIDHttpRequestIntegrationTests.m (hunk 4 above)
Comment (antrix1989) [reply IdentityCore/tests/integration/MSIDHttpRequestIntegrationTests.m]: Removed, it is not actually needed here.

**Pull request merged**